### PR TITLE
Singleton

### PR DESCRIPTION
**Pull request by martinmilev**
Ready for review.

### CODE REVIEW

File: test/com/clouway/singleton/SingletonTest.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * @author Martin Milev <martinmariusmilev@gmail.com>
+ */
+public class SingletonTest {
+
+  @Test
+  public void happyPath() throws Exception {
+    Singleton singleton = Singleton.getInstance();
+    assertThat("com.clouway.singleton.Singleton", is(singleton.getClass().getName()));
```
Comment (mgenov) [inline test/com/clouway/singleton/SingletonTest.java]: Test is wrong. You need to check that there is only one instance.

Comment (mgenov) [inline test/com/clouway/singleton/SingletonTest.java]: The following test: 
```java
A ainstance = new A();
A bisntance = new A(); 
assertThat("package.of.a.A", is(equalTo(ainstance.getClass.getName()); 
assertThat("package.of.a.A", is(equalTo(binstance.getClass.getName()); 
```
verifies that class names of instancea and instanceb are equal to: "package.of.a.A". 

The role of singleton is to have only one instance.

File: src/com/clouway/singleton/Singleton.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.clouway.singleton;
+
+/**
+ * @author Martin Milev <martinmariusmilev@gmail.com>
+ */
+public class Singleton {
```
Comment (mgenov) [inline src/com/clouway/singleton/Singleton.java]: Is this class thread-safe ?

Comment (martinmilev) [inline src/com/clouway/singleton/Singleton.java]: It is not thread-safe. Should I test it with different threads?

Comment (mgenov) [inline src/com/clouway/singleton/Singleton.java]: Why it's not thread safe? 

No I don't want tests about this.

Comment (martinmilev) [inline src/com/clouway/singleton/Singleton.java]: While a thread creates a singleton, another thread could start creating another singleton while the `instance == null` .

Comment (mgenov) [inline src/com/clouway/singleton/Singleton.java]: Ok

File: test/com/clouway/singleton/SingletonTest.java (hunk 1 above)
Comment (martinmilev) [inline test/com/clouway/singleton/SingletonTest.java]: Is that a valid test?
`assertThat(singleton1, is(equalTo(singleton2)));`

Comment (mgenov) [inline test/com/clouway/singleton/SingletonTest.java]: What equalTo does?

Comment (martinmilev) [inline test/com/clouway/singleton/SingletonTest.java]: Checks if the expected object is equal to the actual object.
If it is wrong I cloud use sameInstance() or something else.

Comment (mgenov) [inline test/com/clouway/singleton/SingletonTest.java]: The problem is that you should know which one you have to use.

Singleton states that there should be only one instance, so you have to
test that only single instance is returned.

Comment (martinmilev) [inline test/com/clouway/singleton/SingletonTest.java]: Fixed.

**Comment by mgenov** [comment]
LGTM !

**Pull request merged**